Sources/SwiftRefactor/ConvertZeroParameterFunctionToComputedProperty.swift

```diff
@@ -0,0 +1,69 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if swift(>=6)
+public import SwiftSyntax
+#else
+import SwiftSyntax
+#endif
+
+public struct ConvertZeroParameterFunctionToComputedProperty: SyntaxRefactoringProvider {
+  public static func refactor(syntax: FunctionDeclSyntax, in context: ()) -> VariableDeclSyntax? {
+    guard syntax.signature.parameterClause.parameters.isEmpty,
+      let body = syntax.body
+    else { return nil }
+
+    let variableName = PatternSyntax(
+      IdentifierPatternSyntax(
+        leadingTrivia: syntax.funcKeyword.trailingTrivia,
+        identifier: syntax.name
+      )
+    )
+
+    let paramsTrailingTrivia = syntax.signature.parameterClause.rightParen.trailingTrivia
+    let colonLeadingTrivia = paramsTrailingTrivia == .space ? [] : paramsTrailingTrivia
+
+    var variableType: TypeAnnotationSyntax?
+
+    if let returnClause = syntax.signature.returnClause {
+      variableType = TypeAnnotationSyntax(
+        colon: .colonToken(
+          leadingTrivia: colonLeadingTrivia,
```

Comment:
I think it would be good practice to also retain the leading trivia from the arrow.

```suggestion
          leadingTrivia: triviaFromParameters + returnClause.arrow.leadingTrivia,
```